File: api/src/main/java/com/sinergy/chronosync/dto/request/PaginatedUserRequestDTO.java
package com.sinergy.chronosync.dto.request;

import com.sinergy.chronosync.model.user.UserRole;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.List;

/**
 * Paginated user request DTO.
 */
@Getter
@Setter
@RequiredArgsConstructor
@AllArgsConstructor
public class PaginatedUserRequestDTO extends BasePaginationRequest {

	private String firstName;
	private String lastName;
	private String uniqueIdentifier;
	private String username;
	private List<UserRole> roles;
	private Long firmId;
}

File: gui/src/dtos/requests/PaginatedUserRequestDTO.ts
import { PageableRequest } from '../../models/BaseEntity';
import { UserRole } from '../../models/user/UserRole';

/**
 * Paginated user request DTO for search operations.
 */
export interface PaginatedUserRequestDTO extends PageableRequest {
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  uniqueIdentifier?: string;
  roles?: UserRole[];
}
